### PR TITLE
[behaviorManager] look to partner during pointing behaviour

### PR DESCRIPTION
**Pull request by sockchinglow**
Enabled the iCub to look to human partner midway through the pointing action. Ready for testing.

### CODE REVIEW

**Comment by Tobias-Fischer** [comment]
Hi @sockchinglow, can you please add another commit in this branch replacing tabs by spaces? Also, please change your text editor to use spaces by default :).

File: main/src/modules/reactiveLayer/behaviorManager/src/pointing.cpp [hunk 1]
```diff
@@ -8,28 +8,50 @@ void Pointing::configure() {
...
+
+	iCub->say("I could point to the " + obj_name);
+	Time::delay(2.0);
+	// yInfo() << "About to point the " + obj_name;
+	// return;
+
+	Object* obj = iCub->opc->addOrRetrieveEntity<Object>(obj_name);
+	string sHand = "right";
+	if (obj->m_ego_position[1] < 0)
+		sHand = "left";
```
Comment (Tobias-Fischer) [inline main/src/modules/reactiveLayer/behaviorManager/src/pointing.cpp]: Hi @pattacini,
I saw this check in UPF's code all over the place, and I think it is not necessary. Is ARE smart enough to point with the proper hand by itself, i.e. can we remove passing `bHand` to `iCub->point`?

**Comment by Tobias-Fischer** [comment]
Also, could you please change the looking behavior such that the iCub also looks at the partner when saying `I could point to the x`?
I.e. the iCub first looks at the agent, then says `I could point to the x`, then points, then looks at the partner again, and finally says `Do you know this is a x`?

File: main/src/modules/reactiveLayer/behaviorManager/src/pointing.cpp (hunk 1 above)
Comment (pattacini) [inline main/src/modules/reactiveLayer/behaviorManager/src/pointing.cpp]: Yes, indeed.

Comment (Tobias-Fischer) [inline main/src/modules/reactiveLayer/behaviorManager/src/pointing.cpp]: Cool.
@clement-moulin-frier @sockchinglow @jypuigbo can you please change your modules to not pass `bHand` to the `ARE` subsystem? That shortens the code by quite a bit.

Thanks,
Tobi

**Comment by sockchinglow** [comment]
I am making the changes and will test them some more before the branch is ready to merge with the master.

**Comment by Tobias-Fischer** [comment]
Hi,
Please DO NOT merge this commit. Something must have gone wrong when rebasing this branch.
@pattacini: Can you please help @sockchinglow out on this? I'm on vacation and unfortunately won't have time to do the proper rebasing.

**Comment by pattacini** [comment]
@Tobias-Fischer now the rebase should be ok.

@sockchinglow I've just correctly rebase `looking_agent_fromopc` on top of `master`. Therefore, I'd really suggest you to delete your local branch and track this one from scratch, to be clean.

Here are the commands:

```sh
>> git checkout master
>> git branch -d looking_agent_fromopc
>> git fetch origin
>> git branch looking_agent_fromopc --track origin/looking_agent_fromopc
>> git checkout looking_agent_fromopc
```

**Comment by pattacini** [comment]
@sockchinglow I don't know what went wrong with your previous operation, but I think we could do the rebase together next time to let you practice with it.

For that we can use https://ide.c9.io/pattacini/wysiwyd

**Comment by Tobias-Fischer** [comment]
Thanks Ugo!

**Comment by sockchinglow** [comment]
@pattacini I tried to rebase this branch on top of the master before pushing so I can test from my desktop, but I must have done something wrong...again. I'm so sorry (@Tobias-Fischer too)! I'll delete this branch and create a new one to work on.

However, please note that I have yet to test the new pointing command on the robot so don't use the module just yet.

**Comment by Tobias-Fischer** [comment]
Hi Sock, don't be sorry! All good :).
No need to create a new branch, just execute the commands written by @pattacini above.

**Comment by pattacini** [comment]
Never mind @sockchinglow, git might be quite tough to learn at the beginning but once accustomed to it you'll appreciate its versatility :smirk: 

Let me know when you'll be ready to merge this PR, so that we can do the proper rebase together.

**Comment by pattacini** [comment]
Let me close this PR and open a brand new one to merge into year-2-demo.

File: main/src/modules/reactiveLayer/behaviorManager/src/pointing.cpp [hunk 2]
```diff
@@ -10,28 +10,59 @@ void Pointing::configure() {
...
     string obj_name = sensation->get(id).asList()->get(1).asString();
-    
+
+    iCub->opc->checkout();
+    yInfo() << "[pointing] : opc checkout";
+    list<Entity*> lEntities = iCub->opc->EntitiesCache();
+    string aName = "";
+    for (auto& entity : lEntities)
+    {
+        if (entity->entity_type() == "agent")
```
Comment (Tobias-Fischer) [inline main/src/modules/reactiveLayer/behaviorManager/src/pointing.cpp]: Hi @sockchinglow, can you please check for `entity->m_present==1.0` as well? Thanks!

File: main/src/modules/reactiveLayer/behaviorManager/src/pointing.cpp [hunk 3]
```diff
@@ -10,28 +10,59 @@ void Pointing::configure() {
...
-    string sHand = "right";
-    if (obj->m_ego_position[1]<0) 
-        sHand = "left";
+    //string sHand = "right";
+    //if (obj->m_ego_position[1] < 0)
+    //    sHand = "left";
+
+    //Bottle bHand(sHand);
+
+    //bool succeeded = iCub->point(obj_name, bHand);
```
Comment (Tobias-Fischer) [inline main/src/modules/reactiveLayer/behaviorManager/src/pointing.cpp]: Can you please delete these comments? They won't ever be uncommented, and thus just make the code harder to read.

**Pull request closed without merging**